conjure-java-jersey-server/src/main/java/com/palantir/conjure/java/server/jersey/RemoteExceptionMapper.java

```diff
@@ -54,64 +54,35 @@
-        }
-        return builder.build();
-    }
-
-    private ErrorType mapErrorType(RemoteException exception) {
-        Status status = Status.fromStatusCode(exception.getStatus());
-
-        if (status.getStatusCode() == 401) {
-            log.info(
-                    "Encountered a remote unauthorized exception."
-                            + " Mapping to a default unauthorized exception before propagating",
-                    SafeArg.of("errorInstanceId", exception.getError().errorInstanceId()),
-                    SafeArg.of("errorName", exception.getError().errorName()),
-                    SafeArg.of("statusCode", status.getStatusCode()),
-                    exception);
-
-            return ErrorType.UNAUTHORIZED;
-        } else if (status.getStatusCode() == 403) {
+        if (exception.getStatus() == 401 || exception.getStatus() == 403) {
             log.info(
-                    "Encountered a remote permission denied exception."
-                            + " Mapping to a default permission denied exception before propagating",
+                    "Encountered a remote exception",
                     SafeArg.of("errorInstanceId", exception.getError().errorInstanceId()),
                     SafeArg.of("errorName", exception.getError().errorName()),
-                    SafeArg.of("statusCode", status.getStatusCode()),
+                    SafeArg.of("statusCode", exception.getStatus()),
                     exception);
 
-            return ErrorType.PERMISSION_DENIED;
+            return Response.status(exception.getStatus())
+                    .type(MediaType.APPLICATION_JSON)
+                    .entity(SerializableError.builder()
+                            .errorCode(exception.getError().errorCode())
+                            .errorName(exception.getError().errorName())
+                            .errorInstanceId(exception.getError().errorInstanceId())
+                            .build())
+                    .build();
         } else {
-            // log at WARN instead of ERROR because this indicates an issue in a remote server
             log.warn(
                     "Encountered a remote exception. Mapping to an internal error before propagating",
                     SafeArg.of("errorInstanceId", exception.getError().errorInstanceId()),
                     SafeArg.of("errorName", exception.getError().errorName()),
-                    SafeArg.of("statusCode", status.getStatusCode()),
+                    SafeArg.of("statusCode", exception.getStatus()),
                     exception);
 
-            return ErrorType.INTERNAL;
+            ServiceException propagatedException = new ServiceException(ErrorType.INTERNAL, exception);
+            return Response.status(propagatedException.getErrorType().httpErrorCode())
+                    .type(MediaType.APPLICATION_JSON)
+                    .entity(SerializableError.forException(propagatedException))
```

Comment:
This pattern is used extensively in the Undertow exception handler:
https://github.com/palantir/conjure-java/blob/develop/conjure-java-undertow-runtime/src/main/java/com/palantir/conjure/java/undertow/runtime/ConjureExceptions.java

I slightly prefer this approach because it delegates some of the tricker bits to centralized static factory methods.